sbt-tzdb/src/main/scala/io/github/sbt/tzdb/SbtTzdb.scala

```diff
@@ -54,10 +63,20 @@ object TzdbPlugin extends AutoPlugin {
   )
   override val projectSettings    =
     Seq(
+      generatedSourceDialect := {
+        val neededFlags = Set("-indent", "-new-syntax", "-source:future")
+        if (scalaBinaryVersion.value == "3" && neededFlags.forall(scalacOptions.value.contains))
+          Dialect.Scala3Future
```

Comment:
What do you mean by `Scala3Future`?

Reply:
Scala 3 code compiled with the `-source:future` flag which among other things:

- [A more specialized binding on the left hand side than the type on the right side](https://dotty.epfl.ch/docs/reference/changed-features/pattern-bindings.html#bindings-in-pattern-definitions)
- [Extending classes without the open modifier causes error](https://dotty.epfl.ch/docs/reference/other-new-features/open-classes.html#migration)
- [Old syntax for vararg splices causes error](https://dotty.epfl.ch/docs/reference/changed-features/vararg-splices.html)
- [Operator used infix without infix modifier cause errors](https://dotty.epfl.ch/docs/reference/changed-features/operators.html)
- [Wilcard arguments in Types using the old syntax (aka `_`) produce an error](https://dotty.epfl.ch/docs/reference/changed-features/wildcards.html)
- [Wildcard imports with old syntax (aka `_` and `=>`) cause errors](https://dotty.epfl.ch/docs/reference/changed-features/imports.html)

Although I conflated that with the indentation based syntax and the new control flow syntax. I can make more combinations of flags but that easily becomes unmanagable.

Reply:
Right, thanks for the explantion